MWPW-139487 - Fix for Gnav scroll issue with Landscape Orientation on devices

Deva309 opened the pull request:
<!-- Before submitting, please review all open PRs. -->
## Description
On the landscape orientation for mobile, the user cannot access (scroll to) the lower part of the mega menu.

Resolves: [MWPW-139487](https://jira.corp.adobe.com/browse/MWPW-139487)

## Screenshots:
Before:
<img width="1800" alt="Screenshot 2024-06-14 at 2 00 42 PM" src="https://github.com/adobecom/milo/assets/41122199/e12a3262-2132-4580-bcca-66425b3ed30e">

After:
<img width="1800" alt="Screenshot 2024-06-14 at 2 00 34 PM" src="https://github.com/adobecom/milo/assets/41122199/5c94ece4-834a-4b78-8343-657703ac2153">

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/devashish/wam?martech=off
- After: https://mwpw-139487--milo--deva309.hlx.page/drafts/devashish/wam?martech=off

overmyheadandbody commented on libs/blocks/global-navigation/utilities/menu/menu.css:
```diff
@@ -5,8 +5,7 @@
 .feds-menu-headline[aria-expanded = "true"] + .feds-menu-items {
-  display: flex;
-  flex-direction: column;
+  display: inline-block;
```
This will cause issues in the footer

<img width="428" alt="Screenshot 2024-06-17 at 12 09 57" src="https://github.com/adobecom/milo/assets/11267498/27f36477-66cf-45c4-88f6-f0a3bee38f21">

Deva309 replied:
Fixed.
Replaced with `display: block;`.

overmyheadandbody commented on libs/blocks/global-navigation/global-navigation.css:
```diff
@@ -551,7 +551,8 @@ header.global-navigation {
 
   .feds-menu-container {
-    min-width: 100%;
+    max-width: 1200px;
```
There's already a variable for this value, `--feds-minWidth-menu`

Deva309 replied:
Added `--feds-minWidth-menu` variable for width.

overmyheadandbody reviewed (changes requested):
I'm not sure this is the right direction. The aim is to adapt the height, yet width and layout properties are affected instead, leading to additional issues. I added the highlighted properties from the screenshot to the `feds-popup` styles starting at `900px` and things seem to work as expected. Will need a lot more testing, but this just impacts the height of the relevant element.

<img width="293" alt="Screenshot 2024-06-17 at 12 27 39" src="https://github.com/adobecom/milo/assets/11267498/1e8d3939-705f-49db-9811-b77fd3cdc3fd">

Deva309 commented:
> I'm not sure this is the right direction. The aim is to adapt the height, yet width and layout properties are affected instead, leading to additional issues. I added the highlighted properties from the screenshot to the `feds-popup` styles starting at `900px` and things seem to work as expected. Will need a lot more testing, but this just impacts the height of the relevant element.
> 
> <img alt="Screenshot 2024-06-17 at 12 27 39" width="293" src="https://private-user-images.githubusercontent.com/11267498/340273788-1e8d3939-705f-49db-9811-b77fd3cdc3fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2ODcyMDIsIm5iZiI6MTcxODY4NjkwMiwicGF0aCI6Ii8xMTI2NzQ5OC8zNDAyNzM3ODgtMWU4ZDM5MzktNzA1Zi00OWRiLTk4MTEtYjc3ZmQzY2RjM2ZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE4VDA1MDE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiODgxZmMwNjI2OGY2Njg2NDc0OWVlODYyMWE4NGViZThjMjRiNjUwNTFkNDM5ZWNmOWJiOTJhMjI5MmU2MDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Gg8S-4B6EFn8MC7_znFaYPYnK2Xf5hDYNTv7LYAp9eY">

@overmyheadandbody 
`flex-direction` does not have any impact now because we updated `display` property from `flex` to `block`.
I have verified all the viewports with this change.Also, height impact only on `feds-popup` as we are calculating based on viewport height and this will only applied when viewport height is less than `mega-menu` height.
I have verified all the viewports with this change. But yeah this change needs to be tested properly.

overmyheadandbody reviewed (commented):
I still don't agree with this approach. Again, you want to adapt behavior on the Y axis, yet this PR makes changes to the X axis. The screenshot I provided yesterday with another way of achieving scroll inside popups only should be reconsidered. That only affects the Y axis and makes changes to a single element.

Deva309 commented:
> I still don't agree with this approach. Again, you want to adapt behavior on the Y axis, yet this PR makes changes to the X axis. The screenshot I provided yesterday with another way of achieving scroll inside popups only should be reconsidered. That only affects the Y axis and makes changes to a single element.

@overmyheadandbody 
I understand your point about the unintended problems caused by changing the width and layout. This affected other sections, as you pointed out in footer. So, I've undone those changes and now I'm only fixing the height of the `megamenu` and adding scrolling just for that element. However, we still need horizontal scrolling (scroll to X axis) because we have
promo that get cut off on the right side, similar to what you might see on a Photoshop page below:

<img width="1800" alt="Screenshot 2024-06-18 at 3 40 02 PM" src="https://github.com/adobecom/milo/assets/41122199/d274715d-f969-4c63-b361-ddc90fe87df3">

That's why I included `overflow: auto` in this scenario.

overmyheadandbody reviewed (approved):
Thanks for updating the approach!

prativas22 commented:
Validation done on the below url :
https://main--cc--adobecom.hlx.live/creativecloud/file-types/image?milolibs=mwpw-139487--milo--deva309
https://main--cc--adobecom.hlx.live/creativecloud?milolibs=mwpw-139487--milo--deva309
https://main--cc--adobecom.hlx.live/creativecloud/file-types/image?milolibs=mwpw-139487--milo--deva309&akamaiLocale=tr
vertical scrolling to the lower part of Mega menu
![image](https://github.com/adobecom/milo/assets/169240751/e5912a9b-d9a5-4ef0-8173-371471c3fd7e)
![image](https://github.com/adobecom/milo/assets/169240751/94595172-0703-4c4e-b31b-cfad2ce60f8d)
horizontal scrolling to the right part of Mega menu
![image](https://github.com/adobecom/milo/assets/169240751/80cb8f07-a0ae-4b95-9dbd-4a0017d89f1d)
![image](https://github.com/adobecom/milo/assets/169240751/83db22e7-dd7e-465a-a234-00708eee0b79)
